### PR TITLE
DSPDC-114 Add "submit transfer" functionality

### PR DESCRIPTION
**Pull request by danxmoran**
Sorry for the huge diff 😬 I reworked a bunch of the existing APIs as I encountered pain points.

### CODE REVIEW

File: agents/template/src/main/scala/org/broadinstitute/transporter/TransporterAgent.scala [hunk 1]
```diff
@@ -21,10 +23,19 @@ import scala.concurrent.ExecutionContext
...
-abstract class TransporterAgent extends IOApp with CirceEntityDecoder {
+abstract class TransporterAgent extends IOApp.WithContext with CirceEntityDecoder {
 
   private val logger = Slf4jLogger.getLogger[IO]
 
+  // Use a single thread for the streams run loop.
+  // Kafka Streams is an inherently synchronous API, so there's no point in
+  // keeping multiple threads around.
+  override val executionContextResource: Resource[SyncIO, ExecutionContext] = {
+    val allocate = SyncIO(Executors.newFixedThreadPool(1))
```
Comment (jacarey) [inline agents/template/src/main/scala/org/broadinstitute/transporter/TransporterAgent.scala]: can use newSingleThreadExecutor if you want I think

File: manager/db/changesets/01-init-queues-table.xml [hunk 2]
```diff
@@ -4,11 +4,15 @@
...
 
-    <changeSet id="1" author="danxmoran">
+    <changeSet id="1" author="danmoran">
+
         <createTable tableName="queues">
-            <column name="name" type="varchar(250)">
+            <column name="id" type="uuid">
                 <constraints primaryKey="true" />
             </column>
+            <column name="name" type="varchar(250)">
```
Comment (jacarey) [inline manager/db/changesets/01-init-queues-table.xml]: 250 for queue name seems excessive? or are we adding some long unique identifier here?

File: manager/src/main/scala/org/broadinstitute/transporter/db/DbClient.scala [hunk 3]
```diff
@@ -123,7 +156,9 @@ object DbClient {
     implicit val schemaGet: Get[QueueSchema] = pgDecoderGet
     implicit val schemaPut: Put[QueueSchema] = pgEncoderPut
 
-    /** Check if the client can interact with the backing DB. */
+    // Glue for the "safe" UUID type our lib generates.
+    implicit val fuuidPut: Put[FUUID] = Put[UUID].contramap(FUUID.Unsafe.toUUID)
```
Comment (jacarey) [inline manager/src/main/scala/org/broadinstitute/transporter/db/DbClient.scala]: FU UID 🤔

File: manager/src/main/scala/org/broadinstitute/transporter/db/DbClient.scala [hunk 4]
```diff
@@ -137,30 +172,72 @@ object DbClient {
...
+          FUUID.randomFUUID[IO].map(id => FUUID.Unsafe.toUUID(id) -> transfer)
+        }
+        _ <- insertRequest(queueId, requestUuid, transfersById.map(_._1))
+          .transact(transactor)
+      } yield {
+        (requestUuid, transfersById)
+      }
+
+    override def deleteTransferRequest(id: UUID): IO[Unit] =
+      sql"""delete from transfer_requests where id = $id""".update.run.void
```
Comment (jacarey) [inline manager/src/main/scala/org/broadinstitute/transporter/db/DbClient.scala]: transfer has an fk to requests right?  Does this need to cascade?

Comment (danxmoran) [reply manager/src/main/scala/org/broadinstitute/transporter/db/DbClient.scala]: Yes, it's set up in the liquibase migrations. [This test](https://github.com/DataBiosphere/transporter/pull/5/files#diff-a7a32a41d1669f65a9ea5a73d73b74d7R119) checks that it actually cascades.

**Pull request merged**